src/Gtk/Avalonia.Gtk/WindowImpl.cs

```diff
@@ -60,20 +60,18 @@ protected override void OnRealized ()
             _imContext.ClientWindow = this.GdkWindow;
         }
 
-        public Size ClientSize
+        public virtual Size ClientSize
         {
             get
             {
                 int width;
                 int height;
                 GetSize(out width, out height);
                 return new Size(width, height);
-            }
-
-            set
-            {
-                Resize((int)value.Width, (int)value.Height);
-            }
+            }	
+            set 
```

Comment:
Previously, the set would call `Gtk.Window.Resize` on `Gtk.Window`. Commenting out this line will make resizing work, but will break sizing for Popups because they use the same implementation.

My change makes this property virtual, and overrides it in the `PopupImpl` and calls `Gtk.Window.Resize` there.

As far as removing the empty setter, I considered it, but then I'd need to modify `ITopLevelImpl` to remove the setter, breaking `PopupImpl`. 

I'm wondering if a more correct fix would be to call something better than `Gtk.Window.Resize` for setting the `ITopLevelImpl.ClientSize` property.